src/misc/src/MonteCarloQuadrature.C

```diff
@@ -0,0 +1,66 @@
+//-----------------------------------------------------------------------bl-
+//--------------------------------------------------------------------------
+//
+// QUESO - a library to support the Quantification of Uncertainty
+// for Estimation, Simulation and Optimization
+//
+// Copyright (C) 2008-2017 The PECOS Development Team
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the Version 2.1 GNU Lesser General
+// Public License as published by the Free Software Foundation.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc. 51 Franklin Street, Fifth Floor,
+// Boston, MA  02110-1301  USA
+//
+//-----------------------------------------------------------------------el-
+
+#include <queso/MonteCarloQuadrature.h>
+#include <queso/UniformVectorRV.h>
+#include <queso/GslVector.h>
+#include <queso/GslMatrix.h>
+
+namespace QUESO
+{
+  template <class V, class M>
+  MonteCarloQuadrature<V,M>::MonteCarloQuadrature( const VectorSubset<V,M> & domain,
+                                                   unsigned int n_samples )
+    : MultiDQuadratureBase<V,M>(domain)
+  {
+    // We'll just uniformly sample over the domain to retrieve the quadrature
+    // points.
+    UniformVectorRV<V,M> uniform_rv("MonteCarloQuadrature_", //prefix
+                                    this->m_domain);
+
+    const BaseVectorRealizer< V, M > & 	realizer = uniform_rv.realizer();
+
+    this->m_positions.resize(n_samples,NULL);
+
+    for( unsigned int i = 0; i < n_samples; i++ )
+      {
+        // Create new vector from our vector space
+        // We are taking ownership of this pointer.
+        typename QUESO::SharedPtr<V>::Type domain_vec(domain.vectorSpace().newVector());
+
+        // Draw the new point
+        realizer.realization(*domain_vec);
```

Comment:
You don't use a uniform distribution for the realisations, you use a distribution that isn't improper on an unbounded domain and the weights are divided by PDF values at the points where you have realisations.

Thinking about it, we ought to consider other flavours of Monte Carlo quadrature with various different realiser strategies.  If you know something about the function you're integrating it might influence the choice of realiser.